### DISCUSSION TITLE
⏳ 🔗 Extend Role manager to stop users from seeing the edit section for the "Header", "Edit Single Page" and "Footer" templates

### DISCUSSION THREAD

**Discussion by liv2day** [elementor/elementor#22840] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

The problem is that right now any Admin editing a post / page will see the "Header", "Edit Single Page" and "Footer" edit section in their screen and can accidentally click on it. Then all of the sudden jump into those templates and start editing them. This results into users messing up templates and confusion. Basically only me, the web builder should be able to mess those things up ;).

### Describe the solution you'd like

Extend Role manager to:
1. Not see the "Header", "Edit Single Page" and "Footer" edit block in Elementor when editing a page.
2. Option to disallow seeing and editing templates in the WP menu.

**Comment by jazir555** [comment]
Very much so agreed!

**Comment by marks7777** [comment]
I support that.

**Comment by chrisdarrell** [comment]
Very much agree....  I have clients breaking templates all the time now...  Wasn't an issue until they made it so easy to access by just clicking on it....  I hate that.

**Comment by black-eye** [comment]
Yes, please. This is essential. Even though I need to give the editing rights to regular users, I don't want them to mess with the crucial parts like headers and footers, which usually contain stuff beyond their knowledge.

And to be honest, the transitions between the pages and templates are quite confusing even for me, an experienced developer. Not to mention I find them buggy and it's often impossible to to return back, when you click the edit template button by mistake. So I would welcome to have a possibility to disable it altogether.

**Comment by chrisdarrell** [comment]
Totally agree.  Im surprised that more people are not asking for it.


Sincerely
Chris Raines 
***@***.***

**Comment by phkoon** [comment]
A much needed feature

**Comment by kenbrooks-rd** [comment]
please add this feature! I have clients breaking header & footers; I want them to be able to edit page content but not access any templates

**Comment by githubtoh** [comment]
Agreed. Keep template editing in the top black bar and in the backend.

**Comment by edknewseed** [comment]
Has anyone tried configuring the user roles via Elementor -> Role manager?  Under each user role there is an option checkbox for 'Access to edit content only'.  Not sure if this really prevent the edit of the header/footer nor do I know if that option still gives full access to edit the main content.

**Comment by kenbrooks-rd** [comment]
That option unfortunately is blanket permission; it applies to page and header/footer editing.

**Comment by edknewseed** [comment]
Have you tried using AI like ChatGPT?  I just did it but haven't tried.  Here's the code snippet.

```
function restrict_elementor_templates_editing( $can_edit, $post_id ) {
    if ( current_user_can( 'manage_options' ) ) {
        // Allow Admins to edit everything
        return $can_edit;
    }

    // Get the post type
    $post_type = get_post_type( $post_id );

    // Restrict editing for headers and footers
    if ( $post_type === 'elementor_library' ) {
        $template_type = get_post_meta( $post_id, '_elementor_template_type', true );
        if ( in_array( $template_type, [ 'header', 'footer' ] ) ) {
            return false; // Prevent editing
        }
    }

    return $can_edit;
}
add_filter( 'elementor/can_edit', 'restrict_elementor_templates_editing', 10, 2 );
```

**Reply by ScottMadore** [reply]
Has anyone tried this?

**Comment by ScottMadore** [comment]
This is very much needed.
I just added a page template to a site I am working on to help customize the page.
I was previously only using Elementor full-width. 

When I add a page template, this makes the experience terrible. 
You don't want to go edit the page template, header, and footer every time you edit your page. You just want to edit your content.

This should really be a toggle somewhere. And as requested, have the ability to turn it off for editors/clients.

Thanks. Please consider adding this asap.
